src/Type/WPInterfaceTrait.php

```diff
@@ -157,29 +157,31 @@ protected function get_fields( array $config, TypeRegistry $type_registry ): arr
 			$fields = array_merge( $fields, $diff );
 		}
 
+		// Loop through the fields to inherit undefined config values from the interface.
 		foreach ( $fields as $field_name => $field ) {
 			$new_field = $field;
 
 			if ( ! isset( $new_field['type'] ) ) {
-				if ( isset( $interface_fields[ $field_name ]['type'] ) ) {
-					$new_field['type'] = $interface_fields[ $field_name ]['type'];
-				} else {
+				// If neither the field nor the interface have a type, the field is invalid and should be removed.
+				if ( ! isset( $interface_fields[ $field_name ] ) ) {
 					unset( $fields[ $field_name ] );
```

Comment:
I think this would be a good place for a `graphql_debug()` message but i didnt want to change functionality.